pkg/taskservice/types.go

```diff
@@ -43,22 +83,24 @@ type TaskService interface {
 	// CreateCronTask is similar to Create, but create a task that runs periodically, with the period
 	// described using a Cron expression.
 	CreateCronTask(ctx context.Context, task task.TaskMetadata, cronExpr string) error
-	// QueryTask search tasks according to conditions.
-	QueryTask(...QueryOption) ([]task.Task, error)
-	// CronTasks returns all created cron tasks
-	CronTasks() ([]task.CronTask, error)
+	// Complete task completed.
+	Complete(ctx context.Context, taskRunner string, task task.Task) error
 	// Heartbeat sending a heartbeat tells the scheduler that the specified task is running normally.
 	// If the scheduler does not receive the heartbeat for a long time, it will reassign the task executor
 	// to execute the task. Returning `ErrInvalidTask` means that the Task has been reassigned or has
 	// ended, and the Task execution needs to be terminated immediately。
-	Heartbeat(taskRunner string, tasks ...task.Task) error
+	Heartbeat(ctx context.Context, taskRunner string, tasks ...task.Task) error
+	// QueryTask query tasks by conditions
+	QueryTask(context.Context, ...Condition) ([]task.Task, error)
```

Comment:
Who is the intended user of this QueryTask interface?   Scheduler?

Reply:
Both Scheduler and TaskRunner.  Scheduler use it to query all tasks waiting to be scheduled. TaskRunner query all pending tasks assigned to self.